doc/guides/writing-tests.md

```diff
@@ -209,6 +209,35 @@ const assert = require('assert');
 const freelist = require('internal/freelist');
 ```
 
+### Assertions
+
+When writing assertions, prefer the strict versions:
+
+* `assert.strictEqual()` over `assert.equal()`
+* `assert.deepStrictEqual()` over `assert.deepEqual()`
+
+When using `assert.throws()`, if possible, provide the full error message:
```

Comment:
It may just be terminology I'm unfamiliar with, but without the context of this conversation, I don't know what this would mean. If someone told me, "That's pretty good, but could you make the regular expression more concrete?" ... I would have to ask what they mean.

I think the existing wording in this PR is the best we've managed to come up with yet. I'm OK with leaving it as is and saving any bike-shedding for subsequent revisions, but if we want to tweak it now maybe this?:

>  When using `assert.throws()`, the second argument should be either a regular expression or a function to check that the error received is as expected. If using a regular expression, make the match as specific as possible. Match the entire error message by starting the regular expression with `^` and ending with `$`.